packages/primitives/text-block/src/index.web.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { component, startWithType, mapProps } from 'refun'
+import { Block } from '@primitives/block'
+import { Text } from '@primitives/text'
+import { TBlockProps, TTextBlock } from './types'
+
+export const TextBlock = component(
+  startWithType<TTextBlock>(),
+  mapProps(
+    ({
+      textAlign,
+      snapStart,
+      width,
+      ...textProps
+    }) => {
+      const blockProps = {} as TBlockProps
+
+      if (snapStart) {
+        blockProps.marginTop = -snapStart
+        blockProps.paddingTop = snapStart
+      }
+
+      if (width) {
+        blockProps.width = width
+      }
+
+      if (textAlign) {
+        blockProps.textAlign = textAlign
+      }
+
+      return {
+        blockProps,
+        textProps,
+      }
+    }
+  )
+)(({ blockProps, textProps }) => {
+  return (
+    <Block style={blockProps}>
```

Comment:
Should we maybe use the `<p>` element in web? Which is a text block semantically too 🤔

Reply:
I considered that too and, tbh I'm still divided. My reasons to go for this alternative though are:

1. Parity with React Native: there's no kind of block text element in React Native, so if we go for a `<p>` in web, this primitive will output 2 elements in RN and 1 element in web.

2. The `<Text>` primitive outputs a `<p>` with `display: inline` in web, this would eventually be made as a point for `<Text>` to support `display: block` instead of having an additional primitive in the web.

Reason 2 is weak and I don't think it would stand by itself. But I have a hunch that keeping output parity between RN and Web can save us some headaches growing up. 🤔

Reply:
So for the primitives, *1. Parity with React Native* is not the top prio in my mind. The idea is that the primitives serve as the abstraction layer between the two platforms, so as long as the external API and behavior is uniform, the implementation details are secondary.

That said maybe there's something we can't achieve with just a `<p>`,  but I'll try if possible to keep the semantics clean.

For 2. , I'm not sure I follow. Maybe `@primitives/text` should not output `<p>` at all once we introduce this `text-block`